torch/_inductor/codegen/wrapper.py

```diff
@@ -1086,19 +1163,31 @@ def codegen_device(self, device):
     def make_buffer_allocation(self, buffer):
         from .cpp import DTYPE_TO_ATEN
 
-        # TODO: map layout here
-        device = buffer.get_device()
-        dtype = buffer.get_dtype()
-        shape = tuple(buffer.get_size())
-        stride = tuple(buffer.get_stride())
-        device_str = self.codegen_device
-        return (
-            f"{self.declare}{buffer.get_name()} = {self.namespace}empty_strided("
-            f"{self.codegen_shape_tuple(shape)}, "
-            f"{self.codegen_shape_tuple(stride)}, "
-            f"{self.codegen_device(device)}"
-            f".dtype({DTYPE_TO_ATEN[dtype]})){self.ending}"
-        )
+        output_idx = None
+        for idx, output in enumerate(V.graph.graph_outputs):
+            if isinstance(output, (ir.NoneAsConstantBuffer, ir.ShapeAsConstantBuffer)):
+                continue
+            if buffer == output.data:
+                output_idx = idx
+                break
+        if output_idx is not None and V.graph.aot_mode:
+            # In aot_mode, output buffers are managed by the AOT runtime.
+            return (
+                f"at::Tensor {buffer.get_name()} = outputs[{output_idx}]{self.ending}"
```

Comment:
How do we guarantee the passed-in output buffers are valid? So basically what is our protocol with the runtime?

Reply:
The runtime will first invoke ```AOTInductorModelContainerGetMaxInputShape``` to query the output tensor shapes before allocating the right amount of memory.